### ISSUE TITLE
Implement MeterProvider Options

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#2819]
- [x] `WithResource`
- [x] `WithReader`

Blocked by ~#2800~, ~#2817~, ~#2809~

**Comment by MrAlias** [comment]
Based on prior discussions, it seems like we want to associate views with readers. I'm planing a PR to propose:

```go
func WithResource(res *resource.Resource) Option

func WithReader(r Reader, opts ...view.Option) Option
```

As found [here](https://github.com/open-telemetry/opentelemetry-go/blob/3c905851606ab084d5b58c8377cc4fef42b8a941/sdk/metric/config.go#L49-L65).

**Comment by MrAlias** [comment]
Closed by #2905
